src/components/secure-backup/index.tsx

```diff
@@ -61,92 +76,102 @@ export class SecureBackup extends React.PureComponent<Properties, State> {
     return this.props.backupExists && !this.props.isBackupRecovered;
   }
 
-  renderBackedUpMode() {
-    const { isLegacy } = this.props;
-    const button = isLegacy ? <Button onPress={() => this.props.onGenerate()}>Generate new backup</Button> : null;
+  get isSystemPrompt() {
+    return this.backupStage === BackupStage.SystemPrompt;
+  }
+
+  renderHeader = () => {
     return (
-      <div {...cn('backed-up')}>
-        <p>This session is backing up your keys.</p>
-        <p>This backup is trusted because it has been restored on this session.</p>
-        {this.renderButtons(button)}
+      <div {...cn('header')}>
+        <h3 {...cn('title')}>Account Backup</h3>
+        <IconButton {...cn('close')} Icon={IconXClose} onClick={this.props.onClose} size={40} />
       </div>
     );
-  }
+  };
 
-  renderNoBackupMode() {
+  renderVideoBanner = () => {
     return (
-      <>
-        <p>
-          Your keys are <b>not being backed up from this session.</b>
-        </p>
-        <p>Back up your keys before signing out to avoid losing them.</p>
-        {this.renderButtons(<Button onPress={() => this.props.onGenerate()}>Generate backup</Button>)}
-      </>
+      <div {...cn('video-banner')}>
+        <video
+          {...cn('video')}
+          // TODO: uncomment when env vars updated
+          // src={`${this.props.assetsPath}/E2EEncryption.mp4`}
+          autoPlay
+          loop
+          muted
+        />
+      </div>
     );
-  }
+  };
 
-  renderCreationMode() {
-    return (
-      <>
-        <p>
-          The following recovery key has been generated for you. Click `Copy` to copy it to your clipboard. Store it in
-          a secure place such as a password manager of vault then click `Save Backup` to complete the backup process.
-        </p>
-        {this.props.recoveryKey && <div {...cn('recovery-key')}>{this.props.recoveryKey}</div>}
-        {this.renderButtons(
+  renderBackupContent = () => {
+    const {
+      backupStage,
+      onGenerate,
+      onRestore,
+      onVerifyKey,
+      onClose,
+      onSave,
+      recoveryKey,
+      errorMessage,
+      successMessage,
+    } = this.props;
+
+    switch (backupStage) {
+      case BackupStage.None:
+      case BackupStage.SystemPrompt:
+        const promptProps = { ...(this.isSystemPrompt && { onClose }) };
+        return (
           <>
-            {this.props.recoveryKey && <Button onPress={this.copyKey}>Copy</Button>}
-            <Button onPress={() => this.props.onSave()} isDisabled={!this.state.hasCopied}>
-              Save backup
-            </Button>
-          </>
-        )}
-      </>
-    );
-  }
+            {this.noBackupExists && (
```

Comment:
It feels weird to have a single stage also have extra checks like this. Not sure if these should be separate stages or should there just be a single SystemPrompt component or something?

Reply:
Agree, I think these could be made into separate stages. Will follow up.